### PR TITLE
[10712] Statistics module Fast DDS layer API

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
This PR defines:

* Statistics namespace
* Statistics topic names constants
* Statistics default QoS constants
* Extended DomainParticipant class

Related documentation PR eProsima/Fast-DDS-docs#244

### CODE REVIEW

File: include/fastdds/statistics/dds/domain/DomainParticipant.hpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DomainParticipant.hpp
+ *
+ */
+
+#ifndef _STATISTICS_DDS_DOMAIN_DOMAINPARTICIPANT_HPP_
```
Comment (MiguelCompany) [inline include/fastdds/statistics/dds/domain/DomainParticipant.hpp:20]: The following suggestion applies to all public headers. At least for new files, we should mimic the full path

```suggestion
#ifndef _FASTDDS_STATISTICS_DDS_DOMAIN_DOMAINPARTICIPANT_HPP_
```

File: include/fastdds/statistics/dds/domain/DomainParticipant.hpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+ * Class DomainParticipant: extends standard DDS DomainParticipant class to include specific methods for the Statistics
+ * module
+ * @ingroup STATISTICS_MODULE
+ */
+class DomainParticipant : public eprosima::fastdds::dds::DomainParticipant
+{
+public:
+
+    /**
+     * @brief This operation enables the Statistics DataWriter
```
Comment (MiguelCompany) [inline include/fastdds/statistics/dds/domain/DomainParticipant.hpp:47]:
```suggestion
     * @brief This operation enables a Statistics DataWriter
```

File: include/fastdds/statistics/dds/domain/DomainParticipant.hpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+     * RETCODE_INCONSISTENT_POLICY if the DataWriterQos provided is inconsistent,
+     * RETCODE_OK if the DataWriter has been created or if it has been created previously,
+     * and RETCODE_ERROR otherwise
+     */
+    RTPS_DllAPI ReturnCode_t enable_statistics_datawriter(
+            const std::string& topic_name,
+            const eprosima::fastdds::dds::DataWriterQos& dwqos);
+
+    /**
+     * @brief This operation disables the Statistics DataWriter
```
Comment (MiguelCompany) [inline include/fastdds/statistics/dds/domain/DomainParticipant.hpp:61]:
```suggestion
     * @brief This operation disables a Statistics DataWriter
```

File: include/fastdds/statistics/topic_names.hpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+#define _STATISTICS_TOPIC_NAMES_HPP_
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+namespace statistics {
+
+//! Statistic topic that reports the write-to-notification latency between any two pairs of matched
+//! DataWriter-DataReader histories
+constexpr char HISTORY_LATENCY_TOPIC[] = "_fastdds_statistics_history2history_latency";
```
Comment (MiguelCompany) [inline include/fastdds/statistics/topic_names.hpp:29]: This suggestion applies to all the constants on this file

```suggestion
constexpr const char* HISTORY_LATENCY_TOPIC = "_fastdds_statistics_history2history_latency";
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/statistics/topic_names.hpp [hunk 5]
```diff
@@ -16,54 +16,54 @@
...
-#ifndef _STATISTICS_TOPIC_NAMES_HPP_
-#define _STATISTICS_TOPIC_NAMES_HPP_
+#ifndef _FASTDDS_STATISTICS_TOPIC_NAMES_HPP_
+#define _FASTDDS_STATISTICS_TOPIC_NAMES_HPP_
 
 namespace eprosima {
 namespace fastdds {
-namespace dds {
 namespace statistics {
+namespace dds {
```
Comment (MiguelCompany) [inline include/fastdds/statistics/topic_names.hpp:25]: I think these constants should be on the `eprosima::fastdds::statistics` namespace (i.e. not `::dds`)

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: include/fastdds/statistics/dds/domain/DomainParticipant.hpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+namespace fastdds {
+namespace statistics {
+namespace dds {
+
+/**
+ * Class DomainParticipant: extends standard DDS DomainParticipant class to include specific methods for the Statistics
+ * module
+ * @ingroup STATISTICS_MODULE
+ */
+class DomainParticipant : public eprosima::fastdds::dds::DomainParticipant
```
Comment (MiguelCompany) [inline include/fastdds/statistics/dds/domain/DomainParticipant.hpp:42]: This class is not meant to be directly constructed, so its default constructor should be `=delete`d

File: include/fastdds/statistics/topic_names.hpp (hunk 5 above)
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/statistics/topic_names.hpp]: Done in 5a672028fdeeea4ea622c9f20f9c8e5e6b741f78

File: include/fastdds/statistics/dds/domain/DomainParticipant.hpp (hunk 6 above)
Comment (JLBuenoLopez-eProsima) [reply include/fastdds/statistics/dds/domain/DomainParticipant.hpp:42]: Done in 80ef7c62e4d554e676f197a8d5a79f55d0577d29

File: test/unittest/statistics/dds/StatisticsEnvironmentVariableTests.cpp [hunk 7]
```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (MiguelCompany) [inline test/unittest/statistics/dds/StatisticsEnvironmentVariableTests.cpp:1]: Looks like this file was accidentally added.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Pull request merged**